Original patch from gentoo: ../../dev/crosstool-NG/ct-ng.trunk/patches/strace/4.5.17/170-arm-syscalls.patch
-= BEGIN original header =-
Author: Michael Abbott <michael.abbott@diamond.ac.uk>
Date:   Wed Oct 22 08:53:42 2008 +0100

    Syscall patch for ARM
    Patch mailed to strace mailing list by Anderson Lizardo, work by
    Bruna Moreira, dated 2008-01-25.  Patch cleaned up against 4.5.17 release.

diff --git a/linux/arm/errnoent1.h b/linux/arm/errnoent1.h
new file mode 100644
index 0000000..a894827
-= END original header =-
diff -durN b/linux/arm/errnoent1.h b/linux/arm/errnoent1.h
--- b/linux/arm/errnoent1.h	1970-01-01 01:00:00.000000000 +0100
+++ b/linux/arm/errnoent1.h	2008-10-24 21:30:13.000000000 +0200
@@ -0,0 +1,2 @@
+/* Our second set comes from the i386 files.  */
+#include "../errnoent.h"
diff -durN b/linux/arm/ioctlent1.h b/linux/arm/ioctlent1.h
--- b/linux/arm/ioctlent1.h	1970-01-01 01:00:00.000000000 +0100
+++ b/linux/arm/ioctlent1.h	2008-10-24 21:30:13.000000000 +0200
@@ -0,0 +1,2 @@
+/* Our second set comes from the i386 files.  */
+#include "../ioctlent.h"
diff -durN b/linux/arm/signalent1.h b/linux/arm/signalent1.h
--- b/linux/arm/signalent1.h	1970-01-01 01:00:00.000000000 +0100
+++ b/linux/arm/signalent1.h	2008-10-24 21:30:13.000000000 +0200
@@ -0,0 +1,2 @@
+/* Our second set comes from the i386 files.  */
+#include "../signalent.h"
diff -durN b/linux/arm/syscallent1.h b/linux/arm/syscallent1.h
--- b/linux/arm/syscallent1.h	1970-01-01 01:00:00.000000000 +0100
+++ b/linux/arm/syscallent1.h	2008-10-24 21:30:13.000000000 +0200
@@ -0,0 +1,7 @@
+/* ARM specific syscalls */
+	{ 5,	0,	printargs,		"SYS_0"			}, /* 0 */
+	{ 5,	0,	printargs,		"breakpoint"		}, /* 1 */
+	{ 5,	0,	printargs,		"cacheflush"		}, /* 2 */
+	{ 5,	0,	printargs,		"usr26"			}, /* 3 */
+	{ 5,	0,	printargs,		"usr32"			}, /* 4 */
+	{ 5,	0,	printargs,		"set_tls"		}, /* 5 */
